config/smart_answers/next_steps_for_your_business.yml

```diff
@@ -0,0 +1,31 @@
+sections:
+  - title: "Business tax"
+    results:
+      - title: "You need to register for corporation tax within 3 months of starting to do business, if you haven’t done so already."
+        url: "/corporation-tax"
+        description: "Register for Corporation Tax"
```

Comment:
Nit: Wondering whether we could flatten the data structure, and add "section_name" as an attribute on a result? This data organisation seem tightly coupled to the view and might make hard to make changes in future. (Not something blocking this PR tho)

Reply:
Wouldn't that mean we'll introduce some redundancy where multiple results are part of the same section? Then if we don't group them under a section how would we render multiple results under a single title?

Reply:
So yup, it'll be more verbose. I find that the nesting structure can make it difficult to read and can make it more complicated to make changes i.e. it's easier to change the "section_name" value rather than move the whole result. This might be further problematic if we need to add another layer of grouping e.g ""what you need to do soon", "you might be interested in" etc etc

(Also further down the line if this data is being published from another system, it's probably unlikely to be structured with the nesting.)

Logic for a view can handle grouping results into sections - 
```ruby
# Create a hash where the key is the section name and value is array of results
grouped_results = results.group_by { |result| results["section_name"] }

# Access them like so...
business_tax_results = grouped_results["Business tax"]

# or...
grouped_results.each do |section_name, results| .... 
```

Reply:
(Again not something blocking this PR, but we can think about in future)

Reply:
I see your point about this data being made available via a publishing system – I'll give it a go at flattening it and grouping based on a string key (for now – we may be able to record these attributes as entities in a distant future)

Reply:
Have updated this as suggested – thanks for expanding on it!